Criterion/Internal.hs

```diff
@@ -111,33 +112,45 @@ runAndAnalyse :: (String -> Bool) -- ^ A predicate that chooses
               -> Benchmark
               -> Criterion ()
 runAndAnalyse select bs = do
-  mbRawFile <- asks rawDataFile
-  (rawFile, handle) <- liftIO $
-    case mbRawFile of
+  mbJsonFile <- asks jsonFile
+  (jsonFile, handle) <- liftIO $
+    case mbJsonFile of
       Nothing -> do
         tmpDir <- getTemporaryDirectory
-        openBinaryTempFile tmpDir "criterion.dat"
```

Comment:
IIUC, `criterion.dat` is used for incrementally writing out the binary report. Are we breaking `--raw` further by removing it?